### ISSUE TITLE
corrupted size vs prev_size

### ISSUE THREAD

**Issue by przem360** [nesbox/TIC-80#823]
Hi!
Today I was trying to run Tic-80 on my Linux Mint 19.1 Cinnamon, but it failed wit a message:

> corrupted size vs. prev_size
> aborted (memory dump)

It also may by problem with my pc because tic-80 was working just fine few weeks ago. Perhaps this is due to some dependency updates?

**Comment by mt12345** [comment]
The same error on Ubuntu MATE 18.04

**Comment by przem360** [comment]
I do not know why, I do not know how... but it works again.
In history.log of my apt I cant see nothing related to TIC-80.
But it works :)

**Comment by nesbox** [comment]
Hmm, can't reproduce...

**Comment by dreamwebspace** [comment]
I had this problem on Ubuntu 18.04, after upgrading to 19.10 it works fine.

**Comment by farvardin** [comment]
I'd suggest to reopen this one.

This problem occurs on one of my computers, and I can't understand why. I can investigate further if you can point me to some tests.

I've made all the updates on my Linux mint 19.3, and it's the same. Some weeks ago it was working ok on the same computer. I have also other computers with the same OS, without any problem on them.

This computer is running Intel graphical Card ("Intel Corporation 4 Series Chipset Integrated Graphics Controller")

I've run it with gdb, and got:

> Reading symbols from tic80...done.
> (gdb) run
> Starting program: /temp/github/TIC-80/build/bin/tic80 
> [Thread debugging using libthread_db enabled]
> Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
> [New Thread 0x7fffe8fd1700 (LWP 16267)]
> [New Thread 0x7fffe87d0700 (LWP 16268)]
> [New Thread 0x7fffe7fcf700 (LWP 16269)]
> [New Thread 0x7fffe77ce700 (LWP 16270)]
> [New Thread 0x7ffff7fa6700 (LWP 16271)]
> [New Thread 0x7ffff7fe0700 (LWP 16272)]
> 
>  TIC-80 tiny computer 0.90.1392-dev Pro
>  http://tic80.com (C) 2020
> corrupted size vs. prev_size
> 
> Thread 1 "tic80" received signal SIGABRT, Aborted.
> __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
> 51	../sysdeps/unix/sysv/linux/raise.c: Aucun fichier ou dossier de ce type.
> (gdb) 
> 

but not other information.

**Comment by nesbox** [comment]
@farvardin investigate please, I can't reproduce it on my side.
Thank you.

**Comment by farvardin** [comment]
I can investigate a bit yes, but I'm not a great developper :D 

Here is what I've tried so far:


## Computer side 

I've run a testdisk with no reported error. I suppose my memory is not faulty.


## TIC-80 versions (from the release tab in github) : 

- The 0.60.3 (32 and 64 bit) version is working, I can launch it without problem on this computer
-  0.70.1, 0.70.6 and 0.80 : same "corrupted size vs. prev_size" error
- 0.80 windows version: I can run it on the same OS with wine. 

but I've tried again the 0.70.1 version after running wine, and I ran once, without problem. I've closed it, ran it again, and it crashed with the corrupted size vs. prev_size error

So I've started this 0.70.1 version several times in a row (8 times), I crashed all the time. I've started again the wine version, which used some memory, closed it, started the 0.70.1 version, it crashed again the first time, but after I've started the wine version again, then the 0.70.1 version (on linux) started correctly!

I remember some months ago, when TIC-80 was working (I suppose I was running a pre-80 version I've compiled myself), it crashed once during start-up, then I closed firefox which freed some memory and I could start TIC-80 again. I thought it was a temporary memory error. It really seldom crashed. Now it's most of the time.

I finally managed to run the 0.80 linux version (a post-0.80 dev version I've compiled myself) after running wine several time + the 0.70.1 version.

It also worked once after I started it 12 times!

The 60.3 version works all the time on this computer.


## Libraries 

I've compared the library used.


basically, what is in addition in the 0.7 version are:   
libGLdispatch.so.0 => /usr/lib/x86_64-linux-gnu/libGLdispatch.so.0
libGL.so.1 => /usr/lib/x86_64-linux-gnu/libGL.so.1
libGLX.so.0 => /usr/lib/x86_64-linux-gnu/libGLX.so.0

and in 0.8 :
libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1
libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6

what is strange is 

libSDL2-2.0.so.0
libvorbis.so.0
libpulse.so.0

from the 0.6 version are no longer present in further releases. I suppose it is handled differently.


## Code

I don't know C very well (it's rather almost nothing). I've just searched for "corrupted size vs prev_size". 
I've found this: https://github.com/pjreddie/darknet/issues/105

Someone said: "This happens when you are using memory which was not allocated to you or when you are using free() twice in a row on a pointer."

But I don't know more.

**Comment by nesbox** [comment]
I added Ubuntu build without SDLGPU lib support, could you please check it?
Download here https://github.com/nesbox/TIC-80/actions/runs/361286385
`tic80-ubuntu-sdl` and `tic80-ubuntu-sdlgpu`

Thank you.

**Comment by farvardin** [comment]
thank you, I'll try this next week: I don't have a direct access to this computer at the moment. Well, I can connect to it, and I've tried to start the usual TIC-80 through ssh / X forwarding, expecting it to crash as before, but I can tell it starts every time, so I suppose the problem is with the display (it's intel driver if I remember well).

**Comment by farvardin** [comment]
> I added Ubuntu build without SDLGPU lib support, could you please check it?

Here is the result of my tests:

### 0.90.1412-dev

#### tic80-ubuntu-sdl.zip

works flawlessly!

Once installed, I could start it 10 times in a row without the "corrupted size vs. prev_size" or any crash.

#### tic80-ubuntu-sdlgpu.zip

I've tested it too, it looks like it's crashing less than previously, but I get the "corrupted size vs. prev_size"crash  maybe 7 times out of 10.

### 0.80.1344 

I've installed again the tic80-v0.80-linux.deb and it's crashing like 9 times out of 10 (roughtly, but it can be more)


In conclusion, without the SDLGPU support, it seems to work fine. What is the benefit of adding it? Without gpu support, it looks like to be as responsive as the with gpu support.

thank you.

**Comment by nesbox** [comment]
CRT monitor effect won't work without `sdlgpu` lib, just use `SDL` version if you don't need it.

**Comment by nesbox** [comment]
I think this crash is a SDL GPU lib issue.
